File: CardMasterSkin/Elements/SERectangle.vb
﻿Imports System.Drawing
Imports System.Drawing.Drawing2D
Imports CardMasterCard.Card
Imports CardMasterSkin.GraphicsElement

Namespace Skins

    Public Class SERectangle
        Inherits SkinElement

        Public Sub New(width As Integer, height As Integer)
            Me.New(0, 0, width, height)
        End Sub

        Public Sub New(x As Integer, y As Integer, width As Integer, height As Integer)
            MyBase.New(x, y, width, height)
        End Sub

        Protected Overrides Function GetGraphicElements(card As Card) As List(Of GraphicElement)

            Dim graphicElementsList As New List(Of GraphicElement)()
            Dim path As New GraphicsPath()

            path.AddRectangle(New Rectangle(X, Y, Width, Height))

            graphicElementsList.Add(New PathElement(path, GetBackground))

            Return graphicElementsList

        End Function

    End Class

End Namespace

File: CardMasterSkin/Elements/SERectangle.cs
using CardMasterCard.Card;
using CardMasterSkin.GraphicsElements;
using CardMasterSkin.Skins;
using System.Collections.Generic;
using System.Drawing;
using System.Drawing.Drawing2D;

namespace CardMasterSkin.Elements
{
    public class SERectangle : SkinElement
    {
        public SERectangle(Skin skin, int width, int height) : this(skin, 0, 0, width, height)
        { }

        public SERectangle(Skin skin, int x, int y, int width, int height) : base(skin, x, y, width, height)
        { }

        protected override List<GraphicElement> GetGraphicElements(Card card)
        {
            var graphicElementsList = new List<GraphicElement>();
            var path = new GraphicsPath();

            path.AddRectangle(new Rectangle(this.X, this.Y, this.Width, this.Height));

            graphicElementsList.Add(new PathElement(path, GetBackground()));

            path = null;

            return graphicElementsList;

        }

    }
}
